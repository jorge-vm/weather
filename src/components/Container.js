import { useState } from "react";

import SearchForm from "./SearchForm";
import ResultsCard from "./ResultsCard";
import config from "../config.json";

const Container = () => {
  const [weather, setWeather] = useState(null);
  const [city, setCity] = useState(null);
  const [error, setError] = useState(null);

  async function getData(url = "") {
    try {
      const response = await fetch(url);
      if (!response.ok) {
        throw new Error("Network response was not ok");
      }
      return response.json();
    } catch (error) {
      console.error(
        "There has been a problem with your fetch operation:",
        error
      );
    }
  }

  const apiCall = async ({ city, units }) => {
    const url = `${config.url}?q=${city}&appid=${config.key}&units=${units}`;
    const request = getData(url);
    const response = await request;
    if (response != null) {
      setWeather(response.main);
      setCity(response.name);
      setError(null);
    } else {
      setError("Something went wrong. Please try again.");
      setWeather(null);
      setCity(null);
    }
  };

  return (
    <div className="container">
      <div className="container p-3">
        <h1>Learn the weather, plan your day!</h1>

        <SearchForm apiCall={apiCall} />
        {error != null && (
          <div className="container p-1">
            <div className="alert alert-danger fade show" role="alert">
              {error}
            </div>
          </div>
        )}
      </div>
      <div className="container p-3">
        {weather !== null && <ResultsCard weather={weather} city={city} />}
      </div>
    </div>
  );
};

export default Container;
