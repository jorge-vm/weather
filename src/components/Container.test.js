import { render, screen } from "@testing-library/react";
import Container from "./Container";

test("renders Container", () => {
  const { container } = render(<Container />);
  expect(container.firstChild).toMatchSnapshot();
});
