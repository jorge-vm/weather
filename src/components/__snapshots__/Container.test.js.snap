// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Container 1`] = `
<div
  class="container"
>
  <div
    class="container p-3"
  >
    <h1>
      Learn the weather, plan your day!
    </h1>
    <form>
      <div
        class="input-group"
      >
        <span
          class="input-group-text"
        >
          City
        </span>
        <input
          class="form-control"
          id="city"
          name="city"
          placeholder="Enter city name, i,e. Seattle"
          type="text"
          value=""
        />
        <select
          class="form-select"
          name="units"
        >
          <option
            value="imperial"
          >
            °F
          </option>
          <option
            value="metric"
          >
            °C
          </option>
        </select>
      </div>
      <br />
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
  <div
    class="container p-3"
  />
</div>
`;
